ec/ecresource/deploymentresource/apm/v2/apm_payload.go

```diff
@@ -117,6 +119,47 @@ func ApmPayload(ctx context.Context, apmObj types.Object, updateResources *model
 	return payload, nil
 }
 
+func objectToApm(ctx context.Context, plan types.Object) (*ApmTF, diag.Diagnostics) {
+	var apm *ApmTF
+
+	if plan.IsNull() || plan.IsUnknown() {
+		return nil, nil
+	}
+
+	if diags := tfsdk.ValueAs(ctx, plan, &apm); diags.HasError() {
+		return nil, diags
+	}
+
+	return apm, nil
+}
+
+func CheckAvailableMigration(ctx context.Context, plan types.Object, state types.Object) (bool, diag.Diagnostics) {
+	apmPlan, diags := objectToApm(ctx, plan)
+	if diags.HasError() {
+		return false, diags
+	}
+
+	apmState, diags := objectToApm(ctx, state)
+	if diags.HasError() {
+		return false, diags
+	}
+
+	// We won't migrate this topology element if 'instance_configuration_id' or 'instance_configuration_version' are
+	// defined on the TF configuration. Otherwise, we may be setting an incorrect value for 'size', in case the
+	// template IC has different size increments
+	if !apmPlan.InstanceConfigurationId.IsUnknown() || !apmPlan.InstanceConfigurationVersion.IsUnknown() {
```

Comment:
Looks like acceptance tests failed because `apmPlan` is `nil` in the test. We need to add the check for `nil` for both `apmPlan` and `apmState`. The same for other components.

Reply:
Added in https://github.com/elastic/terraform-provider-ec/pull/755/commits/a0cc0fc589310cf024c05ea20354af560401baaa. Thanks!